src/twisted/spread/flavors.py

```diff
@@ -109,6 +109,11 @@ def remoteMessageReceived(self, broker, message, args, kw):
         """
         args = broker.unserialize(args)
         kw = broker.unserialize(kw)
+        # Need this to interoperate with Python 2 clients
+        # which may try to send use keywords where keys are of type
+        # bytes.
+        if [key for key in kw.keys() if isinstance(key, bytes)]:
```

Comment:
To be specific about @tardyp's criticism:

1. This examines _every_ key, where it only really needs to examine one.  This test could be much more efficient as `any(key for key in kw.keys() if isinstance(key, bytes))`.
2. This might also be more efficient as an in-place modification of `kw`; additionally, it might be more correct. If we have determined that a certain object is really supposed to be a kwargs map, then we know it ought to have native str types in it, and those should show up in other places if we can apply that heuristic more globally.  It would also avoid copying over the entire dictionary.
3. It does the conversion in the py2 case even though that's arguably _more_ wrong; although `**{u'...': ...}` _works_ in py2, it's only kinda by accident (and the runtime is converting back into `str` under the covers anyway).

It might also be worthwhile to do some up-front negotiation to discover if this is required; two 3-only clients could easily say to each other "hey, we're both py3, no need for that gross bytes/args check, everything should check out" earlier on in the connection.